packages/realm/src/ProgressRealmPromise.ts

```diff
@@ -41,7 +41,7 @@ type OpenBehavior = {
 function determineBehavior(config: Configuration, realmExists: boolean): OpenBehavior {
   const { sync, openSyncedRealmLocally } = config;
   if (!sync || openSyncedRealmLocally) {
-    return { openBehavior: OpenRealmBehaviorType.OpenImmediately };
+    return { openBehavior: Realm.App.Sync.openLocalRealmBehavior.type };
```

Comment:
Since the `openLocalRealmBehavior()` method seem to be the source of truth for the default value, I'm fine with this change. Like you, I'm curious what others think. (Sidenote, would probably be clearer if `default` was in the method name.)

Reply:
I'm not sure if we're talking about the same thing, but `openLocalRealmBehaviour` is actually a static field. Regarding default in the name....I was thinking about something similar too. 
Because from the documentation I understand that `openLocalRealmBehaviour` and `downloadBeforeOpenBehavior` are sort of "presets" that we suggest to developers, so they can use them as they want. Hypothetically, they could even change those, but I don't think that this is a use case that we suggested or even considered

Reply:
I call into question the existence of `openLocalRealmBehavior` and `downloadBeforeOpenBehavior` variables on `Realm.App.Sync`, but maybe that's up for another discussion.  I think we should declare an explicit internal value called `defaultOpenBehavior` and set it explicitly without reference to `openLocalRealmBehavior` and `downloadBeforeOpenBehavior`.  Then it's clear where the defaults are directly in `ProgressRealmPromise`

Reply:
@takameyer I agree with all the points you've made 😁 
To be honest I think that it, instead of those two, it would make sense to have a variable called `defaultConfiguration` that the developers can change, and that is used, if set, whenever a realm is opened without configuration

Reply:
I think we shouldn't expose the `defaultConfiguration`.  Just keep it internal and use it when the `determineBehavior` is called.

Reply:
In the context of this conversation ☝️ I wanted to remind ourselves that the static property is part of [our public API](https://www.mongodb.com/docs/realm-sdks/js/latest/Realm.App.Sync.html) which we'd like to limit breaking from v11 to v12 🙂

Reply:
True, we do have to keep them, but we don't actually have to use them internally.

Reply:
>  but we don't actually have to use them internally.

I don't get that. Then the semantics of then will change? Which is breaking?

Reply:
@kraenhansen in v11 we never used these exposed types (`openLocalRealmBehavior` `downloadBeforeOpenBehavior`).  They exist in `lib/extensions.js` and it is not clear why.  If you read the code for opening realm in extension, the default behaviors are hard coded and not tied to those static properties.  Therefore, I don't think we should tie v12 to those properties either, unless we have good reason to.

Reply:
Oh ... I assumed the documentation wasn't lying 😞 In that case I would probably vote for deprecating it in v11 and removing it entirely in v12.

Reply:
@kraenhansen Well it would technically be a breaking change if we removed them, so in the spirit of v12 not having any breaking changes, we should probably just deprecate them and have the v13 remove them entirely.

Reply:
Just to add, the semantics of the public APIs (e.g. `openLocalRealmBehavior`) and how we could use it internally (e.g. `defaultLocalOpenRealmConfiguration`) from my understand remain unchanged due to:
https://github.com/realm/realm-js/blob/fa9ca548b4741a0a64fe730a44d36ce51fac35cc/packages/realm/src/app-services/Sync.ts#L194-L203

Reply:
P.S. @papafe, I think the words in the names could be rearranged. E.g. `defaultOpenLocalRealmConfiguration` and `defaultOpenSyncedRealmConfiguration` (I think I'd also vote for `behavior` over `configuration` 😉 )

Reply:
I think this thread has grown enough to warrant a small meeting to finalize what exactly we want to do here.

Reply:
I'll tap out for now and trust that y'all know how to reasonably resolve this 🙂